src/Modifier/Product/CategoryModifier.php

```diff
@@ -51,9 +54,21 @@ class CategoryModifier extends Modifier
     /**
      * @param Connection      $connection
      * @param TableTranslator $tableTranslator
+     * @param EditionSelector $editionSelector
      */
-    public function __construct(private Connection $connection, private TableTranslator $tableTranslator)
-    {
+    public function __construct(
+        private Connection $connection,
+        private TableTranslator $tableTranslator,
+        EditionSelector $editionSelector,
+    ) {
+        $isEnterprise = $editionSelector->getEdition() === EditionSelector::ENTERPRISE;
+        if ($isEnterprise) {
```

Comment:
I think this should be better
```suggestion
        if ($editionSelector->isEnterprise()) {
```